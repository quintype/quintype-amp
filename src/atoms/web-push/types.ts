export interface WebPushTypes {
  "helper-iframe-url": string;
  "permission-dialog-url": string;
  "service-worker-url": string;
  "service-worker-scope"?: string;
}
