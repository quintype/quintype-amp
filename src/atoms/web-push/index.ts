import { WebPush } from "./web-push";
export { WebPush };
