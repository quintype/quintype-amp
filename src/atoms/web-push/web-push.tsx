import React, { Fragment } from "react";
import { WebPushTypes } from "./types";
import { Helmet } from "react-helmet";

export const WebPush = (props: WebPushTypes) => {
  return (
    <Fragment>
      <Helmet>
        <script
          async={undefined}
          custom-element="amp-web-push"
          src="https://cdn.ampproject.org/v0/amp-web-push-0.1.js"
        />
      </Helmet>
      <amp-web-push layout="nodisplay" {...props} />
    </Fragment>
  );
};
