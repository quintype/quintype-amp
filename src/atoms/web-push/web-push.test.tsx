import React from "react";
import { WebPush } from "./web-push";
import { shallow } from "enzyme";

const wrapper = shallow(
  <WebPush
    layout="nodisplay"
    helper-iframe-url="https://foo.com/a"
    permission-dialog-url="https://foo.com/b"
    service-worker-url="https://foo.com/c"
  />
);

describe("WebPush", () => {
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should pass all props down to amp-web-push", () => {
    expect(wrapper.find("amp-web-push").prop("layout")).toBe("nodisplay");
    expect(wrapper.find("amp-web-push").prop("helper-iframe-url")).toBe("https://foo.com/a");
    expect(wrapper.find("amp-web-push").prop("permission-dialog-url")).toBe("https://foo.com/b");
    expect(wrapper.find("amp-web-push").prop("service-worker-url")).toBe("https://foo.com/c");
  });
});
