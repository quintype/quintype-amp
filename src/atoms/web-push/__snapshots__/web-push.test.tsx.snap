// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebPush should render 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      custom-element="amp-web-push"
      src="https://cdn.ampproject.org/v0/amp-web-push-0.1.js"
    />
  </HelmetWrapper>
  <amp-web-push
    helper-iframe-url="https://foo.com/a"
    layout="nodisplay"
    permission-dialog-url="https://foo.com/b"
    service-worker-url="https://foo.com/c"
  />
</Fragment>
`;
